content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/showhidedialogfields/source/showhidedialogfieldstabs.js

```diff
@@ -0,0 +1,128 @@
+      });
+    });
+
+  });
+
+  $(document).on("change", "[data-cq-dialog-dropdown-checkbox-showhide]", function() {
+    showHide($(this));
+  });
+
+  function showHide(el) {
+    // get the selector to find the target elements. it is stored as
+    // data-attribute
+    // cqDialogDropdownCheckboxShowhideTarget
+    var target = el.data('cqDialogDropdownCheckboxShowhideTarget');
+    var checkboxValue = '';
+    var dropdownValue = '';
+
+    // check if the changed element is the drop-down or the check-box
+    // and get the values accordingly
+    if ($(el).is("coral-select")) {
+      dropdownValue = getDropdownValue(el);
+      checkboxValue = getCheckboxValue(el.closest('coral-panel-content').find('coral-checkbox'));
+    } else if ($(el).is("coral-checkbox")) {
+      dropdownValue = getDropdownValue(el.closest('coral-panel-content').find('coral-select'));
+      checkboxValue = getCheckboxValue(el);
+    }
+
+    // make sure all target elements are hidden.
+    hideAllTargetElements(target);
+
+    // unhide target elements based on the target values
+    $(target).each(function() {
+      if (shouldBeVisible($(this), dropdownValue, checkboxValue)) {
+        hideElement($(this), false);
+      }
+    });
+  }
+
+  function getDropdownValue(dropdownElement) {
+    return dropdownElement.val();
+  }
+
+  function getCheckboxValue(checkBoxElement) {
+    // is check-box checked?
+    var checked = checkBoxElement.prop('checked');
+    // get the selected value
+    // if check-box is not checked, we set the value to empty string
+    return checked ? checkBoxElement.val() : '';
+  }
+
+  // make sure all target elements are hidden.
+  function hideAllTargetElements(target) {
+    $(target).each(function() {
+      hideElement($(this), true);
+    });
+  }
+
+  /**
+   * Checks if the element should be visible based on selected dropdown value
+   * and checkbox value
+   */
+  function shouldBeVisible($elem, dropdownValue, checkboxValue) {
+    if ($elem.is('[data-dropdownshowhidetargetvalue]') && $elem.is('[data-checkboxshowhidetargetvalue]')) {
+      return $elem.attr('data-dropdownshowhidetargetvalue').indexOf(dropdownValue) >= 0 && $elem.attr('data-checkboxshowhidetargetvalue') === checkboxValue;
+    } else if ($elem.is('[data-dropdownshowhidetargetvalue]')) {
+      return $elem.attr('data-dropdownshowhidetargetvalue').indexOf(dropdownValue) >= 0;
+    } else if ($elem.is('[data-checkboxshowhidetargetvalue]')) {
+      return $elem.attr('data-checkboxshowhidetargetvalue') === checkboxValue;
+    }
+    return false;
+  }
+
+  /**
+   * Hides/unhides the element
+   */
+  function hideElement($elem, hide) {
+    var $field_wrapper = $elem.closest('.coral-Form-fieldwrapper');
+    var tab_panel = $elem.parent().parent("coral-panel[role='tabpanel']");
+    var tab_label_id = $(tab_panel).attr('aria-labelledby');
+
+    if (hide) {
+      // If target is a container, hides the container
+      $elem.addClass('hide');

```

Comment:
is it necessary to hide both `$elem` and $field_wrapper`?

Reply:
It covers the case where one puts multiple fields inside a container element and shows and hides the container instead of doing that per field. For this case, jquery will not select any element for $field_wrapper but still a jquery object will be returned. However, $field_wrapper.addClass('hide') will not lead to any errors.